### DISCUSSION TITLE
ROOT requires matching Xcode (13.0 or less, or 13.1)

### DISCUSSION THREAD

**Discussion by henryiii** [Homebrew/discussions#2466] [Everyday usage]
The ROOT formula is broken (see https://root-forum.cern.ch/t/problem-with-root-on-macos-error-module-darwin-uses-additional-module-map/47443 ) if you have Xcode 13.1 installed; the current build requires 13.0 or earlier. There also has not been a rebuild for macOS 12 (the off by 1 Xcode versions are irritating). This brings me to my questions:

* How do we get a Montery bottle for ROOT (which has several dependencies)? I guess a dependency is stuck? How would I tell?
* Is there a way to rebuild the Big Sur bottle with Xcode 13.1? AFAICT, this is breaking users on Big Sur who have updated Xcode. This also affects Apple Silicon.

**Comment by carlocab** [comment]
Root isn't bottled on Monterey because `gts` and `graphviz` are not yet bottled. On ARM Monterey, `librsvg` and `gcc` + dependents are not yet bottled too. You can check this by doing
```
brew unbottled --tag=arm64_monterey root
```

As for rebuilding with the new Xcode, a `revision` bump should suffice, but this will probably break things for users still on Xcode 12.5 (who presumably things still work for at the moment). We could just rebottle it too so it won't force an upgrade for users who still have a working install.

**Reply by jackfrost373** [reply]
Is there any update on this?

**Reply by carlocab** [reply]
What kind of update are you looking for?

**Reply by jackfrost373** [reply]
I was wondering what the status is regarding a new ROOT bottle that works with xcode 13.1 (and specifically for an ARM target)

**Reply by carlocab** [reply]
I don't know, since I don't use ROOT. However, you can probably work out what you want to find out by looking at the formula history: https://github.com/Homebrew/homebrew-core/commits/master/Formula/root.rb As an alternative (or even in addition) to that, you can try `brew install root` yourself to see if it works.

**Reply by henryiii** [reply]
6.26.00 is coming pretty soon. I missed the last ROOT meeting (early this year), so I don't know the exact status, but "early next year" was the original target.

I'm 90% sure this has already been rebuilt for newest Xcode's, though. I'm up to date and it works, and it claims that I poured it (`brew install root --build-from-source` always worked).

**Reply by jackfrost373** [reply]
thanks!

**Reply by henryiii** [reply]
https://github.com/Homebrew/homebrew-core/pull/96175

**Comment by carlocab** [comment]
By the way, @henryiii, what do you make of the existing Xcode dependency in the `root` formula? I tried deleting that line and `brew test root` worked on a CLT-only system. It did complain a bunch of times about this, though:
```
warning: no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk'
```

**Reply by henryiii** [reply]
I asked about this, it needs to get clang from somewhere, but nobody was sure about it being required.
